app/scripts/controllers/transactions/index.js

```diff
@@ -727,6 +718,29 @@ export default class TransactionController extends EventEmitter {
       this.txStateManager.setTxStatusConfirmed(txId);
       this._markNonceDuplicatesDropped(txId);
 
+      const { submittedTime } = txMeta;
+      const { blockNumber } = txReceipt;
+      const metricsParams = { gas_used: gasUsed };
+      const completionTime = await this._getTransactionCompletionTime(
+        blockNumber,
+        submittedTime,
+      );
+
+      if (completionTime) {
+        metricsParams.completion_time = completionTime;
+      }
+
+      if (txReceipt.status === '0x0') {
+        metricsParams.status = 'failed on-chain';
+        // metricsParams.error = TODO: figure out a way to get the on-chain failure reason
```

Comment:
Do we intend to have this kept in?

Reply:
Yes this will be a followup after some exploration

Reply:
Yeah its TODO to add later